Sources/App/Models/Version.swift

```diff
@@ -59,7 +59,7 @@ final class Version: Model, Content {
     var publishedAt: Date?
 
     @Field(key: "reference")
-    var reference: Reference?
+    var reference: Reference
```

Comment:
The are some other fields in `Version` that could be made non-optional at the same time, essentially the ones we assign when we create a `Version` in `getIncomingVersions`:

```swift
            return try Version(package: package.model,
                               commit: revInfo.commit,
                               commitDate: revInfo.date,
                               reference: ref,
                               url: url)
```

But making them non-optional would make using `Version` in tests a bit more awkward, because we'd need to fill these in everywhere or create a convenience initialiser with some sort of defaults.

Therefore I left it to `reference`, because it's the one we actually use the most and which has the most impact on cleaning up our code.

Reply:
Actually, I'll give it a try to see what the impact is. It's quite tempting to clean up at least `commit` and `commitDate`, which are closely related to `reference`.

Reply:
Turns out it's not too awkward if we introduce a convenience init that we use in tests only where we stub in some defaults for these newly required fields.